### PR TITLE
Issue 1 create an entry

### PR DESCRIPTION
**Pull request by louis-zhangxa**
https://user-images.githubusercontent.com/114542582/199300515-b9623d1e-f670-4b5c-aa3c-08a28011f11c.mov

https://user-images.githubusercontent.com/114542582/199300638-758c73c5-392f-4652-9399-3cb1897e2600.mov

### CODE REVIEW

**Comment by louis-zhangxa** [comment]
Screen Shot 2022-11-01 at 10 52 36 AM

**Comment by louis-zhangxa** [comment]
Screen Shot 2022-11-01 at 10 55 13 AM

File: css/styles.css [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  .user-entry {
+    width: 70%;
+    margin: auto;
+  }
+
+  .flex {
+    display: flex;
+  }
```
Comment (codingcodymiller) [inline css/styles.css]: What are you using this for? Why not just use the `row` class?

File: css/styles.css [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  .header {
+    width: 70%;
+    margin: auto;
+  }
+
+  .user-entry {
+    width: 70%;
+    margin: auto;
+  }
```
Comment (codingcodymiller) [inline css/styles.css]: why are you setting `width` directly? you should be using `column-` classes to regulate element's size, and using the `container` class to restrict the size of the page as a whole

File: index.html [hunk 3]
```diff
@@ -14,11 +14,36 @@
...
   <link rel="stylesheet" href="css/layout.css">
   <link rel="stylesheet" href="css/styles.css">
+  <link rel="preconnect" href="https://fonts.googleapis.com">
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+  <link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@400;600&display=swap" rel="stylesheet">
 </head>
 <body>
   <header>
+    <div class="header">
+      <h3>Code Journel</h3>
```
Comment (codingcodymiller) [inline index.html]: You misspelled "journal". Also, you should have elements with the `container`, `row`, and `column` classes in here.

File: index.html [hunk 4]
```diff
@@ -14,11 +14,36 @@
...
+            <input required type="URL" name="PhotoURL" id="PhotoURL">
+          </div>
+        </div>
+        <label for="Notes">Notes</label>
+        <textarea required name="Notes" id="Notes" cols="20" rows="10"></textarea>
+        <div class="form-actions">
+          <button type="submit" id="submit">SAVE</button>
+        </div>
+      </form>
+    </div>
```
Comment (codingcodymiller) [inline index.html]: Where are your layout classes? I gave you a bunch of them in `layout.css`

Comment (codingcodymiller) [reply index.html]: this layout needs to be done using proper layout classes.

File: js/main.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+var url = $photoUrl.addEventListener('input', function (event) {
+  url = event.target.value;
+});
+var title = $Title.addEventListener('input', function (event) {
+  title = event.target.value;
+});
+var notes = $Notes.addEventListener('input', function (event) {
+  notes = event.target.value;
+});
```
Comment (codingcodymiller) [inline js/main.js]: You shouldn't do this on each `input` event, you should just collect these values when the `submit` event occurs in the `submit` event handler.

File: js/main.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+}
+);
+
+function submitNote(event) {
+  var userNote = {
+    Title: title,
+    PhotoUrl: url,
+    Notes: notes
+  };
+  data.entries.push(userNote);
```
Comment (codingcodymiller) [inline js/main.js]: You should be adding to the beginning, not the end.

File: js/main.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  data.nextEntryId++;
```
Comment (codingcodymiller) [inline js/main.js]: You did not add an id to the `userNote` entry object.

File: js/main.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+function string(event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('javascript-local-storage', dataJSON);
+}
+
+$submit.addEventListener('click', submitNote);
```
Comment (codingcodymiller) [inline js/main.js]: This should be a `submit` event listener on the `form` element, not a `click` event listener on the submit button. Please review `javascript-forms` for a refresher on how to handle this correctly.

File: js/main.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    Notes: notes
+  };
+  data.entries.push(userNote);
+  data.nextEntryId++;
+}
+
+function string(event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('javascript-local-storage', dataJSON);
+}
```
Comment (codingcodymiller) [inline js/main.js]: This should be in `data.js`, in the same file as the variable it is preserving.

File: js/main.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  var userNote = {
+    Title: title,
+    PhotoUrl: url,
+    Notes: notes
+  };
+  data.entries.push(userNote);
+  data.nextEntryId++;
+}
+
+function string(event) {
```
Comment (codingcodymiller) [inline js/main.js]: `string` is a **_terrible_** function name. how about `storeDataInLocalStorage`?

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/data.js [hunk 11]
```diff
@@ -6,3 +6,11 @@ var data = {
...
   nextEntryId: 1
 };
+
+function storeDataInLocalStorage(event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('javascript-local-storage', dataJSON);
+}
+
+var $submit = document.querySelector('form');
+$submit.addEventListener('submit', storeDataInLocalStorage);
```
Comment (codingcodymiller) [inline js/data.js]: This should be a `beforeunload` event listener, not a `submit` event listener on the `form`. LocalStorage is for saving data for future sessions, which means it only makes sense to update its value right before this session ends. It does not need to be updated every time the form is submitted

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**